bfss3/bfss3.go

```diff
@@ -265,42 +264,55 @@ func (b *s3Bucket) Copy(ctx context.Context, src, dst string) error {
 }
 
 // Close implements bfs.Bucket.
-func (*s3Bucket) Close() error { return nil }
+func (*bucket) Close() error { return nil }
 
 // --------------------------------------------------------
 
 type writer struct {
 	*os.File
 
 	ctx    context.Context
-	bucket *s3Bucket
+	bucket *bucket
 	name   string
 	opts   *bfs.WriteOptions
 
 	closeOnce sync.Once
 }
 
-func (w *writer) Close() error {
-	var err error
+func (w *writer) Discard() error {
+	err := context.Canceled
+	w.closeOnce.Do(func() {
+		// Delete tempfile in the end
+		fname := w.Name()
+		defer os.Remove(fname)
+
+		// Close tempfile
+		err = w.File.Close()
+	})
 
+	return err
+}
+
+func (w *writer) Commit() error {
+	err := context.Canceled
```

Comment:
Why? (curious)

It should be overridden, but why it's init-ed as Canceled?

Reply:
to make sure it fails with some error when called after Discard